onnx/defs/nn/defs.cc

```diff
@@ -102,19 +102,19 @@ void convPoolShapeInference(
       int input_dims_size = static_cast<int>(n_input_dims);
       for (int i = 0; i < input_dims_size; ++i) {
         int64_t residual = 0;
-        if (strides[i] > 1) {
+        int64_t stride = strides[i];
+        if (stride > 1) {
           if (!input_shape.dim(2 + i).has_dim_value()) {
             continue;
           }
-          residual =  input_shape.dim(2 + i).dim_value();
-          while (residual > 0) {
-            residual -= strides[i];
+          residual = input_shape.dim(2 + i).dim_value();
+          while (residual >= stride) {
+            residual -= stride;
           }
         }
-        int64_t total_pad = residual == 0 ? kernel_shape[i] - strides[i] : kernel_shape[i] + residual;
-        if (total_pad < 0) {
-          fail_shape_inference("Stride is bigger than Kernel shape");
-        }
+        int64_t total_pad = residual == 0 ? kernel_shape[i] - stride : kernel_shape[i] - residual;
+        if (total_pad < 0)
```

Comment:
LGTM.

Just 1 naive question - do we really need this mutation ? Even if total pads is negative, in reality, this means just ignoring a few values in the raw data ("unpad" a few values) right ? Also, I think it should not affect the output dim value when we use the formula - floor((in_dim - kernel_shape + total_pad) / s) + 1. Does this seem reasonable ?

Reply:
I checked the corner cases and I think probably there is no impact if you leave total_pad as negative value, but for simplicity and clarity, it's better to keep it as non-negative value.

Reply:
@linkerzhang Please review this PR